lib/mapObjects/CObjectClassesHandler.cpp

```diff
@@ -586,15 +590,15 @@ std::vector<ObjectTemplate> AObjectTypeHandler::getTemplates(si32 terrainType) c
 		return filtered;
 }
 
-boost::optional<ObjectTemplate> AObjectTypeHandler::getOverride(si32 terrainType, const CGObjectInstance * object) const
+boost::optional<const ObjectTemplate*> AObjectTypeHandler::getOverride(si32 terrainType, const CGObjectInstance * object) const
```

Comment:
boost optional for pointer is excessive. Just return nullptr for invalid situations

Reply:
It depends. The thing clearly shows that it can fail and return nothing and with boost::optional you hardly can accidently forget about it.

Reply:
you could store optional `reference_wrapper`